Lib/test/test_pathlib.py

```diff
@@ -2026,6 +2026,23 @@ def test_replace(self):
         self.assertEqual(os.stat(r).st_size, size)
         self.assertFileNotFound(q.stat)
 
+    def test_move(self):
+        P = self.cls(BASE)
+        p = P / 'fileA'
+        size = p.stat().st_size
+        # Replacing a non-existing path.
+        q = P / 'dirA' / 'fileAA'
+        replaced_p = p.move(q)
+        self.assertEqual(replaced_p, q)
+        self.assertEqual(q.stat().st_size, size)
+        self.assertFileNotFound(p.stat)
+        # Replacing another (existing) path.
+        r = rel_join('dirB', 'fileB')
+        replaced_q = q.move(r)
+        self.assertEqual(replaced_q, self.cls(r))
+        self.assertEqual(os.stat(r).st_size, size)
+        self.assertFileNotFound(q.stat)
+
```

Comment:
Suggestions for tests:

- Also test moving a file into an existing _directory_
- Also test symlinks
- Also mock out `os.rename()` so that you hit the copying code

Reply:
@barneygale I added a test  for moving a file into an existing directory
The other tests will follow soon.

Reply:
Mock `os.rename()` added.

Reply:
Also added a test for symlink.